### DISCUSSION TITLE
`useNodesState` should have generics for NodeData

### DISCUSSION THREAD

**Discussion by so2liu** [xyflow/xyflow#1944] [General]
`useNodesState` and `useEdgesState` should have generics for data.

Current code in `useNodesEdgesState` is 

`export const useNodesState = createUseItemsState<Node, NodeChange>(applyNodeChanges as ApplyChanges<Node, NodeChange>);`

It should be something like this:

`export const useNodesState<T> = createUseItemsState<Node<T>, NodeChange>(applyNodeChanges as ApplyChanges<Node, NodeChange>);`

( I am not good at TypeScript, only a hope as user)

**Comment by moklick** [comment]
Thanks! This is fixed in v10.0.0-next.49. Please open an issue for these kind of things :)
